### PR TITLE
[fix] Closure pointers are unstable #20

### PR DESCRIPTION
**Pull request by lookfirst**

### CODE REVIEW

**Comment by lookfirst** [comment]
Ok, I just tested this in my own codebase and it is working now.

File: src/use-localstorage.ts [hunk 1]
```diff
@@ -46,31 +46,24 @@ export function useLocalStorage<TValue = string>(key: string, initialValue?: TVa
...
+  }, [key]) as EventListener;
 
   useEffect(() => {
     // The custom storage event allows us to update our component 
     // when a change occurs in localStorage outside of our component
-    window.addEventListener(
-      LocalStorageChanged.eventName,
-      (e: any) => onLocalStorageChange(e as LocalStorageChanged<TValue>)
-    );
+    window.addEventListener(LocalStorageChanged.eventName, onLocalStorageChange));
```
Comment (jharrilim) [inline src/use-localstorage.ts]: Extra parenthesis on L54 I believe.

File: src/use-localstorage.ts [hunk 2]
```diff
@@ -46,31 +46,24 @@ export function useLocalStorage<TValue = string>(key: string, initialValue?: TVa
...
     if (initialValue)
       writeStorage(key, initialValue);
 
     return () => {
-      window.removeEventListener(
-        LocalStorageChanged.eventName,
-        (e: any) => onLocalStorageChange(e as LocalStorageChanged<TValue>)
-      );
-      window.removeEventListener('storage', e => onLocalStorageChange(e));
+      window.removeEventListener(LocalStorageChanged.eventName, onLocalStorageChange));
```
Comment (jharrilim) [inline src/use-localstorage.ts]: Extra parenthesis

**Comment by lookfirst** [comment]
@jharrilim good eye on the parans, sorry about that!

**Comment by lookfirst** [comment]
🏓 ?

**Comment by lookfirst** [comment]
Forget it.

**Pull request closed without merging**